tesla_ble_mqtt/rootfs/app/run.sh

```diff
@@ -63,49 +67,66 @@ send_key() {
+    bashio::log.error "tesla-control could not send the key; make sure the car is awake and sufficiently close to the bluetooth device. Retrying in $SEND_CMD_RETRY_DELAY""
+    bashio::log.error "Retrying in $SEND_CMD_RETRY_DELAY"
     sleep $SEND_CMD_RETRY_DELAY
   fi
  done 
 }
 
+PRESENCE_BINARY_SENSOR=OFF
 listen_to_ble() {
- bashio::log.green "Listening to BLE"
+ PRESENCE_TIMEOUT=5
+ bashio::log.info "Listening to BLE for presence"
  set +e
- bluetoothctl --timeout 5 scan on | grep $BLE_MAC
+ bluetoothctl --timeout $PRESENCE_TIMEOUT scan on | grep $BLE_MAC
  EXIT_STATUS=$?
  set -e
  if [ $? -eq 0 ]; then
-   bashio::log.green "$BLE_MAC presence detected"
-   mosquitto_pub --nodelay -h $MQTT_IP -p $MQTT_PORT -u "$MQTT_USER" -P "$MQTT_PWD" -t tesla_ble/binary_sensor/presence -m ON
+   bashio::log.info "$BLE_MAC presence detected"
+   if [ $PRESENCE_BINARY_SENSOR == "OFF" ]; then
+     bashio::log.info "Updating topic tesla_ble/binary_sensor/presence ON"
+     mosquitto_pub --nodelay -h $MQTT_IP -p $MQTT_PORT -u "$MQTT_USER" -P "$MQTT_PWD" -t tesla_ble/binary_sensor/presence -m ON
+     PRESENCE_BINARY_SENSOR=ON
+   else
+     bashio::log.debug "Topic tesla_ble/binary_sensor/presence already ON"
+   fi
  else
-   bashio::log.yellow "$BLE_MAC presence not detected or issue in command, retrying now"
-   mosquitto_pub --nodelay -h $MQTT_IP -p $MQTT_PORT -u "$MQTT_USER" -P "$MQTT_PWD" -t tesla_ble/binary_sensor/presence -m OFF
+   bashio::log.warning "$BLE_MAC presence not detected or issue in command, will retry later"
+   if [ $PRESENCE_BINARY_SENSOR == "ON" ]; then
+     bashio::log.info "Updating topic tesla_ble/binary_sensor/presence OFF"
+     mosquitto_pub --nodelay -h $MQTT_IP -p $MQTT_PORT -u "$MQTT_USER" -P "$MQTT_PWD" -t tesla_ble/binary_sensor/presence -m OFF
+     PRESENCE_BINARY_SENSOR=OFF
+   else
+     bashio::log.debug "Topic tesla_ble/binary_sensor/presence already OFF"
+   fi
  fi
 }
 
-bashio::log.green "Sourcing functions"
+bashio::log.yellow "Sourcing functions"
 . /app/listen_to_mqtt.sh
 . /app/discovery.sh
 
-bashio::log.green "Setting up auto discovery for Home Assistant"
+bashio::log.info "Setting up auto discovery for Home Assistant"
 setup_auto_discovery
 
-bashio::log.green "Connecting to MQTT to discard any unread messages"
+bashio::log.info "Connecting to MQTT to discard any unread messages"
 mosquitto_sub -E -i tesla_ble_mqtt -h $MQTT_IP -p $MQTT_PORT -u $MQTT_USER -P $MQTT_PWD -t tesla_ble/+
 
-bashio::log.green "Initialize BLE listening loop counter"
+bashio::log.info "Initialize BLE listening loop counter"
 counter=0
-bashio::log.green "Entering main MQTT & BLE listening loop"
+bashio::log.info "Entering main MQTT & BLE listening loop"
 while true
 do
  set +e
  listen_to_mqtt
- ((counter++))
- if [[ $counter -gt 90 ]]; then
-  bashio::log.green "Reached 90 MQTT loops (~3min): Launch BLE scanning for car presence"
-  listen_to_ble
-  counter=0
+ if [ -z "$BLE_MAC" ]; then
+   ((counter++))
```

Comment:
We may pass an explicit parameter to enable BLE presence scan later.

Reply:
If I understand your comment properly, that is added in #25 using configuration setting `BLE_PRESENCE_ENABLE` (true|false).